airflow/providers/amazon/aws/hooks/emr.py

```diff
@@ -169,16 +172,21 @@ def add_job_flow_steps(
         if wait_for_completion:
             waiter = self.get_conn().get_waiter("step_complete")
             for step_id in response["StepIds"]:
-                waiter.wait(
-                    ClusterId=job_flow_id,
-                    StepId=step_id,
-                    WaiterConfig=prune_dict(
-                        {
-                            "Delay": waiter_delay,
-                            "MaxAttempts": waiter_max_attempts,
-                        }
-                    ),
-                )
+                try:
+                    wait(
+                        waiter=waiter,
+                        max_attempts=waiter_max_attempts,
+                        waiter_delay=waiter_delay,
+                        args={"ClusterId": job_flow_id, "StepId": step_id},
+                        failure_message=f"EMR Steps failed: {step_id}",
+                        status_message="EMR Step status is",
+                        status_args=["Step.Status.State", "Step.Status.StateChangeReason"],
+                    )
+                except AirflowException as ex:
+                    if "EMR Steps failed" in str(ex):
+                        resp = self.get_conn().describe_step(ClusterId=job_flow_id, StepId=step_id)
+                        self.log.error("EMR Steps failed: %s", resp["Step"]["Status"]["StateChangeReason"])
```

Comment:
Not really. In the documentation, there is supposed to be a `FailureDetails` field that should provide details, but I haven't seen that in the actual response body.

Reply:
When I have a failed job it takes 3-4 minutes before the logs appear and the issue is discovered. I wonder if the empty body reflects this as querying too soon when there is not yet information from the logs.

Reply:
the waiter is already querying this endpoint, so you're going to get the same info as what's already printed by the method since you put `"Step.Status.StateChangeReason"` in the status args.
+1 to use the `wait` method, but -1 to this `except`

Reply:
Is there any other source for useful information you know of, @dacort?